pkg/planner/core/preprocess.go

```diff
@@ -1176,6 +1176,10 @@ func checkIndexOptions(isColumnar bool, indexOptions *ast.IndexOption) error {
 		switch indexOptions.Tp {
 		case ast.IndexTypeVector, ast.IndexTypeInverted:
 			// Accepted
+		case ast.IndexTypeFulltext:
```

Comment:
Which field do we use to check the storage that this index uses?

Reply:
The PM has not decided the user behavior of TiCI yet, so it is not in this PR

Reply:
According to latest discussion with PM, all columnar indexes (FTS, Vector, Inverted) will have an "advanced mode" parameter. When advanced mode is set, TiCI will be used (and required). When it is not set, TiFlash will be used (and required).

Advanced mode will be added later, because currently there lacks a syntax to specify parameter for columnar indexes. This will be implemented by @Lloyd-Pottiger